library_generation/post_processing/postprocessing_utilities.sh

```diff
@@ -0,0 +1,132 @@
+function run_owlbot_postprocessor {
+  workspace=$1
+  owlbot_sha=$2
+  repo_metadata_json_path=$3
+  include_samples=$4
+  scripts_root=$5
+  destination_path=$6
+  transport=$7
+  repository_path=$8
+  proto_path=${9}
+
+  repository_root=$(echo "${repository_path}" | cut -d/ -f1)
+
+  # read or infer owlbot sha
+  if [ -z "${owlbot_sha}" ]; then
+    if [ ! -d "${output_folder}/${repository_root}" ];
+    then
+      echo 'no owlbot_sha provided and no repository to infer it from. This is necessary for post-processing' >&2
+      exit 1
+    fi
+    echo "no owlbot_sha provided. Will compute from monorepo's head"
+    owlbot_sha=$(grep 'sha256' "${output_folder}/${repository_root}/.github/.OwlBot.lock.yaml" | cut -d: -f3)
+  fi
+
+  cp "${repo_metadata_json_path}" "${workspace}"/.repo-metadata.json
+
+  # call owl-bot-copy
+  owlbot_staging_folder="${workspace}/owl-bot-staging"
+  mkdir -p "${owlbot_staging_folder}"
+  owlbot_postprocessor_image="gcr.io/cloud-devrel-public-resources/owlbot-java@sha256:${owlbot_sha}"
+
+
+
+  # copy existing pom, owlbot and version files if the source of truth repo is present
+  if [[ -d "${output_folder}/${repository_path}" ]]; then
+    rsync -avm \
+      --include='*/' \
+      --include='*.xml' \
+      --include='package-info.java' \
+      --include='owlbot.py' \
+      --include='versions.txt' \
+      --include='.OwlBot.yaml' \
+      --exclude='*' \
+      "${output_folder}/${repository_path}/" \
+      "${workspace}"
+
+    # for HW libraries, put .OwlBot.yaml file from .github into workspace root
+    if [ -f "${workspace}/.github/.OwlBot.yaml" ]; then
+      cp "${workspace}/.github/.OwlBot.yaml" "${workspace}"
+    fi
+  fi
+
+  echo 'Running owl-bot-copy'
+  pre_processed_libs_folder="${destination_path}/pre-processed"
+  mkdir -p "${pre_processed_libs_folder}/${proto_path}/$(basename "${destination_path}")"
+  find "${destination_path}" -mindepth 1 -maxdepth 1 -type d -not -name 'pre-processed' \
+    -exec cp -pr {} "${pre_processed_libs_folder}/${proto_path}/$(basename "${destination_path}")" \;
+  pushd "${pre_processed_libs_folder}"
+  # create an empty repository so owl-bot-copy can process this as a repo
+  # (cannot process non-git-repositories)
+  git init
+  git commit --allow-empty -m 'empty commit'
+  popd # pre_processed_libs_folder
+
+  docker run --rm \
+    --user $(id -u):$(id -g) \
+    -v "${workspace}:/repo" \
+    -v "${pre_processed_libs_folder}:/pre-processed-libraries" \
+    -w /repo \
+    --env HOME=/tmp \
+    gcr.io/cloud-devrel-public-resources/owlbot-cli:latest \
+    copy-code \
+    --source-repo-commit-hash=none \
+    --source-repo=/pre-processed-libraries \
+    --config-file=.OwlBot.yaml
+
+
+  echo 'running owl-bot post-processor'
+  # run the postprocessor
+  docker run --rm -v "${workspace}:/workspace" --user $(id -u):$(id -g) "${owlbot_postprocessor_image}"
+
+  # get existing versions.txt from downloaded repository
```

Comment:
Why do we need to do this in a separate script? I thought it is an existing feature of owlbot that versions in pom are kept as they are?

Reply:
Indeed owlbot has some logic [here](https://github.com/googleapis/synthtool/blob/d52e638b37b091054c869bfa6f5a9fedaba9e0dd/docker/owlbot/java/src/fix-poms.py#L300-L305) - I'm trying to confirm this is actually meant to update existing poms

Reply:
I tried the whole `versions.txt` from `google-cloud-java` and the pom ended with versions of other libraries as dependencies. I'm trying to trim the versions we need only and get rid of this script call

Reply:
Solution: use `versions.txt` at the root of the owlbot image filesystem. I removed this script and its call